x-pack/plugins/enterprise_search/server/routes/app_search/engines.test.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { RequestHandlerContext } from 'kibana/server';
+import { mockRouter, RouterMock } from '../../../../../../src/core/server/http/router/router.mock';
+import { httpServerMock } from '../../../../../../src/core/server/http/http_server.mocks';
+import { RouteValidatorConfig } from '../../../../../../src/core/server/http/router/validator';
+
+import { registerEnginesRoute } from './engines';
+import { ObjectType } from '@kbn/config-schema';
+
+jest.mock('node-fetch');
+const fetch = jest.requireActual('node-fetch');
+const { Response } = fetch;
+const fetchMock = require('node-fetch') as jest.Mocked<typeof fetch>;
+
+describe('engine routes', () => {
+  describe('GET /api/app_search/engines', () => {
+    const APP_SEARCH_HOST_CONFIG = 'http://localhost:3002';
+    const AUTH_HEADER = 'Basic 123';
+    const PAGE_INDEX = 1;
+    const TYPE = 'indexed';
+
+    let router: RouterMock;
+    const mockResponseFactory = httpServerMock.createResponseFactory();
+
+    const AppSearchAPI = {
```

Comment:
This is me just thinking out loud, so feel free to shoot this down - at first glance this test is pretty intimidating because it starts out with a lot of abstracted/mocked API stuff. Any thoughts to taking advantage of JS hoisting and moving lines 28-83 to the bottom of the file (after line 174/a comment block labeled "Test helpers", maybe?), so that the more human-readable test steps/intentions come first, and abstractions/details come after?

If that's a terrible idea though feel free to veto!

Reply:
I'm fine with this. TBH, I used to do this, and probably prefer it, but at some point I think was dinged on a PR for that.

Reply:
Updated in 038b792ec5c239ce33cd95bc889359cddf5637b0